### PR TITLE
Add support for cron auto update job

### PR DESCRIPTION
**Pull request by davidanthoff**

### CODE REVIEW

File: src/command_selfuninstall.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        Ok(val) => {            
+            std::process::Command::new("schtasks.exe")
+                .args([
+                    "/delete",
+                    "/tn",
+                    &format!("Juliaup self update for WSL {} distribution", val),
+                    "/f",
+                ])
+                .output()
+                .with_context(|| "Failed to remove Windows task for juliaupa.")?;
```
Comment (deoradh) [inline src/command_selfuninstall.rs:20]:
```suggestion
                .with_context(|| "Failed to remove Windows task for juliaup.")?;
```

**Comment by simeonschaub** [comment]
So is the idea that this always gets run by the install script or will this need to be set up manually? Perhaps we can also give this a different name to make clear that this just sets up a cron job and doesn't really install Juliaup itself.

**Comment by davidanthoff** [comment]
My medium term plan is that we move as much stuff as possible from the shell script into `juliaup self install`, and then always run `juliaup self install` at the end of the shell script. That in particular includes all the steps we do in the shell script right now to add `juliaup` to the `PATH`.

But for now I was going to merge this as is, just to move forward, and then we can tweak and extend. I think before we move the shell script stuff, we minimally need to add some interactivity to `juliaup self install` to ask whether users are OK with all of this, and also provide some command line args. I was thinking things like `--enable-background-update=true/false` and `--enable-path-modification=true/false`, and then those could also be specified with the shell script and it would just pass these through.

I think it is OK, though, to have the entire `self *` set of commands not be perfect before we merge them, really the only two deployment methods that I consider production ready is the Windows Store and AUR story, and neither is affected by this.

**Comment by simeonschaub** [comment]
Yes, I think that sounds like a plan. Does `juliaup self install` already get called at the end of the shell script or not yet?

**Comment by davidanthoff** [comment]
> Does `juliaup self install` already get called at the end of the shell script or not yet?

Not yet, and maybe we leave it like that until we're more comfortable that it all works as we want it to?

**Comment by simeonschaub** [comment]
Yes, that's what I was going to say as well.

**Review by simeonschaub — APPROVED** [review]
Would be good to fix the unused import warnings, but otherwise this looks good to me.

**Pull request merged**